go/ql/src/experimental/CWE-525/WebCacheDeception.ql

```diff
@@ -0,0 +1,29 @@
+/*
+ * @name Web Cache Deception
+ * @description A caching system has been detected on the application and is vulnerable to web cache deception. By manipulating the URL it is possible to force the application to cache pages that are only accessible by an authenticated user. Once cached, these pages can be accessed by an unauthenticated user.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 9
+ * @precision high
+ * @id go/web-cache-deception
+ * @tags security
+ *       external/cwe/cwe-525
+ */
+
+import go
+
+from
+  DataFlow::CallNode httpHandleFuncCall, DataFlow::CallNode call, DataFlow::Node predecessor,
+  Method get
+where
+  httpHandleFuncCall.getTarget().hasQualifiedName("net/http", "HandleFunc") and
+  httpHandleFuncCall.getNumArgument() > 1 and
```

Comment:
I don't think this line is needed. How could a call to `HandleFunc` have anything other than 2 arguments?

Reply:
Yes. HandleFunc gets 2 arguments. Should I check for "= 2" or remove the line.

Reply:
Remove the line, please.